packages/payment-processor/src/payment/utils.ts

```diff
@@ -420,3 +421,31 @@ export async function revokeErc20Approval(
   const tx = await signer.sendTransaction(preparedTx);
   return tx;
 }
+
+/**
+ * Format a request we wish to build a payment for.
+ * If the request does not use the meta-pn, it returns it as is.
+ * Otherwise, returns the request formatted with the pn of interest
+ */
+export function getFormattedRequest({
+  request,
+  pnIdentifier,
+}: {
+  request: ClientTypes.IRequestData;
+  pnIdentifier?: string;
+}): ClientTypes.IRequestData {
+  const pn = getPaymentNetworkExtension(request);
+  if (!pn?.id || pn.id !== ExtensionTypes.PAYMENT_NETWORK_ID.META) return request;
+  if (!pnIdentifier) throw new Error('Missing pn identifier');
+
+  const extensionOfInterest: ExtensionTypes.IState | undefined = pn.values[pnIdentifier];
+  if (!extensionOfInterest) throw new Error('Invalid pn identifier');
```

Comment:
`Invalid` is wrong, the ID might be valid but missing from the advanced logic state.

Reply:
I disagree, if the ID is valid then it is in the advanced logic state.

Given a request, in state X. we should consider that the request is correct and does not miss data. 
And the validity on `pnIdentifier` is based on X. 

Otherwise, I can pass any `pnIdentifier` with any request, and it will never be considered invalid.